dune/grid/io/file/printgrid.hh

```diff
@@ -0,0 +1,191 @@
+// -*- tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+// vi: set et ts=4 sw=2 sts=2:
+
+#include <fstream>
+#include <string>
+
+#include <dune/grid/common/mcmgmapper.hh>
+
+namespace {
+
+  template<int dim>
+  struct ElementDataLayout
+  {
+    bool contains (Dune::GeometryType gt)
+    {
+      return gt.dim()==dim;
+    }
+  };
+
+  template<int dim>
+  struct NodeDataLayout
+  {
+    bool contains (Dune::GeometryType gt)
+    {
+      return gt.dim()==0;
+    }
+  };
+
+  // Move a point closer to basegeo's center by factor scale (used for drawing relative to the element)
+  template <typename B, typename C>
+  C centrify (B basegeo, C coords, double scale) {
+    C ret = coords;
+    ret -= basegeo.center();
+    ret *= scale;
+    ret += basegeo.center();
+    return ret;
+  }
+
+  // Add a line to the plotfile from p1 to p2
+  template <typename Coord>
+  void draw_line (std::ofstream &plotfile, const Coord &p1, const Coord &p2, std::string options) {
+    plotfile << "set object poly from ";
+    plotfile << p1[0] << "," << p1[1] << " to ";
+    plotfile << p2[0] << "," << p2[1] << " to ";
+    plotfile << p1[0] << "," << p1[1];
+    plotfile << " " << options << std::endl;
+  }
+
+}
+
+/** \brief Print a grid as a gnuplot for testing and development
+ *  \tparam GridType the type of grid to work with
+ *  \param grid the grid to print
+ *  \param helper an MPIHelper to create unique output file names in parallel case
+ *  \param output_file the base of the output filename
+ *  \param size size of the plot in pixels; increase if plot is too cramped
+ *  \param execute_plot whether to execute gnuplot automatically
+ *  \param png whether to use PNG or SVG as the output format
+ *  \param local_corner_indices whether to show local corner indices
+ *  \param local_intersection_indices whether to show local intersection indices
+ *  \param outer_normals whether to show outer normals of intersections
+ *  Creates a gnuplot (one per process if parallel) showing the grid structure with indices, intersection types etc.
+ */
+template <typename GridType>
+void printgrid (const GridType& grid, const Dune::MPIHelper& helper, std::string output_file = "printgrid",
+                int size = 2000, bool execute_plot = true, bool png = true, bool local_corner_indices = true,
+                bool local_intersection_indices = true, bool outer_normals = true)
+{
+
+  // Create output file
+  output_file = output_file + "_" + std::to_string(helper.rank());
+  std::ofstream plotfile (output_file, std::ios::out | std::ios::trunc);
+  if (!plotfile.is_open()) {
```

Comment:
It would be better to throw an exception here, otherwise the user has no idea that something went wrong